lib/index.js

```diff
@@ -326,15 +326,15 @@ exports.deleteStorage = async (functionCtx, ctx) => {
     debug('Deleted issuer');
 };
 
-exports.createFunction = async (ctx, functionSpecification) => {
+exports.createFunction = async (ctx, functionSpecification, storageSpecification) => {
     let functionCreated = false;
     try {
         // Acquire any additional configuration elements from optional components
         let additionalCfg = {};
 
         // Is storage requested?
-        if (functionSpecification.enableStorage) {
-            additionalCfg = await exports.createStorage(ctx);
+        if (storageSpecification) {
+            Object.assign(additionalCfg, storageSpecification);
```

Comment:
additionalCfg = storageSpecification?

Reply:
I spied a pattern in this function (perhaps intended by the original owner) that it could take N different contexts and mush them into `additionalCfg`. Currently there is only one, but again I presumed a patttern

Reply:
There's no pattern present in this function anymore - at least not around additionalCfg.  I'd probably simplify it on ln333 to `additionalCfg = { ...storageSpecification} };` so you don't need a separate line at all.

Also, it's a little odd having the forced dependency when storageSpecification added to the package, especially if people want to hard-code at a particular version of the add-on-sdk (which they would).  Maybe remove this side effect and require the caller to supply it?  I'm willing to accept an error if it's not present.  If this is a heavily breaking change, you can pass on it and I'll make the change when I do the refactor.

Reply:
Or, middle ground, only specify it if it's not already present.

Reply:
1. createStorage now must be explicitly called from install.js.
2. We just discussed that it does not make sense to automagically add a dependency to add-on-sdk in the package.json from within createFunction

Given the two things above, I believe we also should not be augmenting the configuration of the function specification with storage config within the createFunction, but instead expect this to happen as part of install.js code.

Reply:
Removed `package.json` and configuration augmentation from `createFunction`, it now has to be done explicitly by the caller in `install.js`